python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -417,16 +418,15 @@ def mult(df):
         ]
         chunk_results = [function(chk) for chk in chunks]
 
-        if len(chunk_results) > 0 and cudf.utils.dtypes.is_scalar(
-            chunk_results[0]
-        ):
+        if not len(chunk_results):
+            return self.obj.__class__()
```

Comment:
We may get the index type incorrect here. I.E. if an empty dataframe with a `StringIndex` was passed in, I believe this will spit out a `RangeIndex`.

Reply:
Pandas seems to discard the index type:

```python

In [17]: a = pd.DataFrame([], index=[1.0, 2.0]).iloc[:0]

In [18]: a.index
Out[18]: Float64Index([], dtype='float64')

In [19]: a.groupby([]).apply(lambda x: x.max()).index
Out[19]: Int64Index([], dtype='int64')

In [20]: a = pd.DataFrame([], index=['a', 'b']).iloc[:0]

In [21]: a.index
Out[21]: Index([], dtype='object')

In [22]: a.groupby([]).apply(lambda x: x.max()).index
Out[22]: Int64Index([], dtype='int64')
```